library/std/src/sync/condvar.rs

```diff
@@ -21,11 +21,11 @@ impl WaitTimeoutResult {
     ///
     /// # Examples
     ///
-    /// This example spawns a thread which will update the boolean value and
-    /// then wait 100 milliseconds before notifying the condvar.
+    /// This example spawns a thread which will sleep 20 milliseconds before
+    /// updating the condvar and then notify.
     ///
-    /// The main thread will wait with a timeout on the condvar and then leave
-    /// once the boolean has been updated and notified.
+    /// The main thread will wait with a timeout with only 10 milliseconds on the condvar
+    /// and will leave once timeout
```

Comment:
"with a timeout with"... This could be constructed more like "with a 10 millisecond timeout".

And what is the thread "leaving"? The loop, right? But it could be clearer.